### ISSUE TITLE
Unable to retrieve an unpublished data file.

### ISSUE THREAD

**Issue by famuvie** [IQSS/dataverse-client-r#115]
Please specify whether your issue is about:

 - [x] a suggested code or documentation change, improvement to the code, or feature request

I'm having an issue while trying to access an unpublished data file, which requires an API token. Unfortunately, this makes the following code not reproducible. Let me know if there is a way of making a reproducible example in this case.

The problem is that I cannot use any of the `get_dataframe_by_*` functions, due to an issue with `is_ingested()` which seems unable to find the target file.
However, if I work around `is_ingested()` I can retrieve the data as is shown in the example below.

``` r
library(dataverse)
packageVersion("dataverse")
#> [1] '0.3.10'

get_dataframe_by_id(
  fileid = 12930,
  dataset = "https://doi.org/10.18167/DVN1/8Z1ZI9"
)
#> Error in is_ingested(fileid, ...): File information not found on Dataverse API

# A successful read
server <- Sys.getenv("DATAVERSE_SERVER")
key <- Sys.getenv("DATAVERSE_KEY")
fileid = 12930
query <- list(format = "original")
u_part <- "access/datafile/"
u <- paste0(dataverse:::api_url(server), u_part, fileid)
r <- httr::GET(u, httr::add_headers(`X-Dataverse-key` = key), query = query)
httr::content(r)
#> Rows: 1347 Columns: 1
#> ── Column specification ────────────────────────────────────────────────────────
#> Delimiter: ","
#> chr (1): ;Vache;Date;Commune;eleveur;Troupeau;R_bursa;H_marginatum;I_ricinus...
#> 
#> ℹ Use `spec()` to retrieve the full column specification for this data.
#> ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#> # A tibble: 1,347 × 1
#>    `;Vache;Date;Commune;eleveur;Troupeau;R_bursa;H_marginatum;I_ricinus;H_scupe…
#>    <chr>                                                                        
#>  1 1;200532 2248;26/02/2020;POPOLASCA;;;0;0;0;0;0;0;0;0;0                       
#>  2 2;200531 7320;26/02/2020;CASTIFAO;;;0;0;0;0;0;0;0;0;0                        
#>  3 3;200530 9555;26/02/2020;ZALANA;;;1;0;0;0;0;0;0;0;1                          
#>  4 4;200532 8365;26/02/2020;MOLTIFAO;;;0;0;0;0;0;0;0;0;0                        
#>  5 5;200533 1185;26/02/2020;PIEDIGRIGGIO;;;0;0;0;2;0;0;0;0;2                    
#>  6 6;200532 3312;26/02/2020;CORTE;;;0;0;0;0;0;0;0;0;0                           
#>  7 7;200531 0907;26/02/2020;BORGO;;;0;0;0;0;0;0;0;0;0                           
#>  8 8;200532 2246;26/02/2020;POPOLASCA;;;0;0;0;0;0;0;0;0;0                       
#>  9 9;200530 8506;26/02/2020;CORTE;;;1;0;0;0;0;0;0;0;1                           
#> 10 10;200532 2245;26/02/2020;POPOLASCA;;;0;0;0;0;0;0;0;0;0                      
#> # … with 1,337 more rows
```

``` r
sessionInfo()
#> R version 4.1.2 (2021-11-01)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Linux Mint 20.1
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
#> LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
#> 
#> locale:
#>  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
#>  [5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=en_GB.UTF-8   
#>  [7] LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] rstudioapi_0.13 knitr_1.37      magrittr_2.0.2  rlang_1.0.0    
#>  [5] fastmap_1.1.0   fansi_0.5.0     stringr_1.4.0   styler_1.4.1   
#>  [9] highr_0.9       tools_4.1.2     xfun_0.29       utf8_1.2.2     
#> [13] cli_3.1.0       withr_2.4.2     htmltools_0.5.2 ellipsis_0.3.2 
#> [17] yaml_2.2.2      digest_0.6.29   tibble_3.1.6    lifecycle_1.0.1
#> [21] crayon_1.4.2    purrr_0.3.4     vctrs_0.3.8     fs_1.5.2       
#> [25] glue_1.6.1      evaluate_0.14   rmarkdown_2.11  reprex_2.0.1   
#> [29] stringi_1.7.6   compiler_4.1.2  pillar_1.6.4    backports_1.4.1
#> [33] pkgconfig_2.0.3
```

**Comment by famuvie** [comment]
Ultimately, the problem in `is_ingested()` boils down to `dataverse_search()` not finding the file:

``` r
library(dataverse)
server <- Sys.getenv("DATAVERSE_SERVER")
key <- Sys.getenv("DATAVERSE_KEY")
dataverse_search(id = "datafile_12930", type = "file", server = server, key = key)
#> 0 of 0 results retrieved
#> list()
```

<sup>Created on 2022-02-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by pdurbin** [comment]
Hmm, because the file is in draft, I bet `_draft` would need to be appended like this:

```
id = "datafile_12930_draft"
```

@famuvie do you want to see if you can find your draft file that way with curl? You'll have to pass your API token. Docs on this at https://guides.dataverse.org/en/5.9/api/search.html

@kuriwaki this might also work:

```
entityId:12930
```

An example: https://dataverse.harvard.edu/api/search?q=entityId:3371438

(I'm not sure why I suggested `id` instead of `entityId`. The `id` changes (`_draft` is dropped on publish) but `entityId` stays the same.)

**Comment by famuvie** [comment]
I'm not sure how to pass the API token with curl. I'll check.
In the meanwhile, I tried adding "_draft" to the `dataverse_search()" call and it didn't work.
Also tried to reproduce the web-search using keywords, and the file appears in the results among others from the same data set:
``` r
library(dataverse)
server <- Sys.getenv("DATAVERSE_SERVER")
key <- Sys.getenv("DATAVERSE_KEY")
dataverse_search(id = "datafile_12958_draft", type = "file", server = server, key = key)
#> 0 of 0 results retrieved
#> list()
dataverse_search("tick collections on bovine slaughterhouse", type = "file", server = server, key = key)
#> 9 of 9 results retrieved
#>                                             name type
#> 1       2021- 06 Cirad wood collection index.csv file
#> 2                           Bovine_2017_2019.tab file
#> 3                           Bovine_2020_2021.tab file
#> 4                   Bovine_Dictio_2017_2019.xlsx file
#> 5                    Bovine_Dictio_2020_2021.tab file
#> 6                            Horse_2018_2020.csv file
#> 7                    Horse_Dictio_2018_2020.xlsx file
#> 8                    NIRS_Herbarium_Senegal.xlsx file
#> 9 VMERGE_EntomologicalData_2014_2016_Senegal.tab file
#>                                                    url file_id
#> 1 https://dataverse.cirad.fr/api/access/datafile/11836   11836
#> 2 https://dataverse.cirad.fr/api/access/datafile/12931   12931
#> 3 https://dataverse.cirad.fr/api/access/datafile/12930   12930
#> 4 https://dataverse.cirad.fr/api/access/datafile/12929   12929
#> 5 https://dataverse.cirad.fr/api/access/datafile/12932   12932
#> 6 https://dataverse.cirad.fr/api/access/datafile/12934   12934
#> 7 https://dataverse.cirad.fr/api/access/datafile/12935   12935
#> 8 https://dataverse.cirad.fr/api/access/datafile/11800   11800
#> 9  https://dataverse.cirad.fr/api/access/datafile/3057    3057
#>           published_at              file_type
#> 1 2021-06-17T08:51:24Z Comma Separated Values
#> 2                 <NA>          Tab-Delimited
#> 3                 <NA>          Tab-Delimited
#> 4                 <NA>   MS Excel Spreadsheet
#> 5                 <NA>          Tab-Delimited
#> 6                 <NA> Comma Separated Values
#> 7                 <NA>   MS Excel Spreadsheet
#> 8 2021-07-08T06:43:57Z   MS Excel Spreadsheet
#> 9 2019-04-23T10:58:52Z          Tab-Delimited
#>                                                   file_content_type
#> 1                                                          text/csv
#> 2                                         text/tab-separated-values
#> 3                                         text/tab-separated-values
#> 4 application/vnd.openxmlformats-officedocument.spreadsheetml.sheet
#> 5                                         text/tab-separated-values
#> 6                                                          text/csv
#> 7 application/vnd.openxmlformats-officedocument.spreadsheetml.sheet
#> 8 application/vnd.openxmlformats-officedocument.spreadsheetml.sheet
#> 9                                         text/tab-separated-values
#>   size_in_bytes                              md5 checksum.type
#> 1       3810305 28332201ff49b7ce97e0c4becaaf7b70           MD5
#> 2         31146 1da7fcb2de712aaeef34bea975ff7c1c           MD5
#> 3         69648 688c6fc5f92e6526a3cd158854027e8b           MD5
#> 4         36395 497aedb27e243a4d43eff27b036177fa           MD5
#> 5           753 7b6bdb573e5401b47d6e1bde4e266ca8           MD5
#> 6          4011 58a8bcabc3c1b493c45c911cec89dff6           MD5
#> 7         74023 4ff31701f5be86fa1deea8c74c73a986           MD5
#> 8     405598437 61dbfb13d99d8016cb1e90d8a1c0552f           MD5
#> 9       1277653 bffb45250746114905c8d823f4a3dfa9           MD5
#>                     checksum.value
#> 1 28332201ff49b7ce97e0c4becaaf7b70
#> 2 1da7fcb2de712aaeef34bea975ff7c1c
#> 3 688c6fc5f92e6526a3cd158854027e8b
#> 4 497aedb27e243a4d43eff27b036177fa
#> 5 7b6bdb573e5401b47d6e1bde4e266ca8
#> 6 58a8bcabc3c1b493c45c911cec89dff6
#> 7 4ff31701f5be86fa1deea8c74c73a986
#> 8 61dbfb13d99d8016cb1e90d8a1c0552f
#> 9 bffb45250746114905c8d823f4a3dfa9
#>                                                  dataset_name dataset_id
#> 1                             CIRAD wood collection - Dataset       2406
#> 2                                                      Bovine      12928
#> 3                                                      Bovine      12928
#> 4                                                      Bovine      12928
#> 5                                                      Bovine      12928
#> 6                                                       Horse      12933
#> 7                                                       Horse      12933
#> 8    Near Infra Red Spectrum of Herbarium sample from Senegal      11799
#> 9 Sénégal – Données entomologiques du projet VMERGE 2014-2016       3056
#>      dataset_persistent_id
#> 1 doi:10.18167/DVN1/CDHU51
#> 2 doi:10.18167/DVN1/8Z1ZI9
#> 3 doi:10.18167/DVN1/8Z1ZI9
#> 4 doi:10.18167/DVN1/8Z1ZI9
#> 5 doi:10.18167/DVN1/8Z1ZI9
#> 6 doi:10.18167/DVN1/SPVQYX
#> 7 doi:10.18167/DVN1/SPVQYX
#> 8 doi:10.18167/DVN1/ZETBMN
#> 9 doi:10.18167/DVN1/LINKQD
#>                                                                                                                                                                                                                                                                                                                dataset_citation
#> 1                                                                                                                                                                   Langbour, Patrick; Paradis, Sébastien; Thibaut, Bernard, 2018, "CIRAD wood collection - Dataset", https://doi.org/10.18167/DVN1/CDHU51, CIRAD Dataverse, V4
#> 2                                                                                                                                                                        Bartholomee, Colombine, 2022, "Bovine", https://doi.org/10.18167/DVN1/8Z1ZI9, CIRAD Dataverse, DRAFT VERSION, UNF:6:ov2odYXNktsIbiuwc2MDJQ== [fileUNF]
#> 3                                                                                                                                                                        Bartholomee, Colombine, 2022, "Bovine", https://doi.org/10.18167/DVN1/8Z1ZI9, CIRAD Dataverse, DRAFT VERSION, UNF:6:ov2odYXNktsIbiuwc2MDJQ== [fileUNF]
#> 4                                                                                                                                                                        Bartholomee, Colombine, 2022, "Bovine", https://doi.org/10.18167/DVN1/8Z1ZI9, CIRAD Dataverse, DRAFT VERSION, UNF:6:ov2odYXNktsIbiuwc2MDJQ== [fileUNF]
#> 5                                                                                                                                                                        Bartholomee, Colombine, 2022, "Bovine", https://doi.org/10.18167/DVN1/8Z1ZI9, CIRAD Dataverse, DRAFT VERSION, UNF:6:ov2odYXNktsIbiuwc2MDJQ== [fileUNF]
#> 6                                                                                                                                                                                                                   Bartholomee, Colombine, 2022, "Horse", https://doi.org/10.18167/DVN1/SPVQYX, CIRAD Dataverse, DRAFT VERSION
#> 7                                                                                                                                                                                                                   Bartholomee, Colombine, 2022, "Horse", https://doi.org/10.18167/DVN1/SPVQYX, CIRAD Dataverse, DRAFT VERSION
#> 8                                                                                                                  Taugourdeau SImon; Bastianeli Denis; Svensk Mia; Bonnal Laurent; Diatta Ousmane, 2021, "Near Infra Red Spectrum of Herbarium sample from Senegal", https://doi.org/10.18167/DVN1/ZETBMN, CIRAD Dataverse, V1
#> 9 Fall, Assane Gueye; Biteye, Biram; Ciss, Mamadou; Seck, Momar Talla; Apolloni, Andrea; Fall, Moussa; Tran, Annelise; Gimonneau, Geoffrey, 2019, "Sénégal – Données entomologiques du projet VMERGE 2014-2016", https://doi.org/10.18167/DVN1/LINKQD, CIRAD Dataverse, DRAFT VERSION, UNF:6:E4qG20NvrCTE/jwa/yQhTg== [fileUNF]
#>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                description
#> 1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     <NA>
#> 2 On this document, there is only the tick data collection of tick between 2017 and 2019.\n\nSome information about variable :\n\n- ID : Date slaughterhouse visit + last four numbers of cow identifier\n- code boucle : last four numbers of cow identifier\n- Date : date of slaughterhouse visit\n- Commune : origin cow municipalitu\n- Eleveur : origin cow breeder\n- Troupeau : municipality and breeder\n- H_marginatus : number of *H_marginatus* collected\n- R_bursa : number of *R_bursa* collected\n- I_ricinus : number of *I_ricinus* collected\n- H_scupense : number of *H_scupense* collected\n- B_annulatus : number of *B_annulatus* collected\n- R_sanguineus : number of *R_sanguineus* collected\n- H_punctata : number of *H_punctata* collected.
#> 3   On this document, there is only the tick data collection between 2020 and 2021.\n\nSome information about variable :\n\n- Vache : Identifier of cow (10 digits)\n- Date : date of slaughterhouse visit\n- Commune : origin cow municipality\n- Eleveur : origin cow breeder\n- Troupeau : municipality and breeder\n- H_marginatus : number of *H_marginatus* collected\n- R_bursa : number of *R_bursa* collected\n- I_ricinus : number of *I_ricinus* collected\n- H_scupense : number of *H_scupense* collected\n- B_annulatus : number of *B_annulatus* collected\n- R_sanguineus : number of *R_sanguineus* collected\n- H_punctata : number of *H_punctata* collected.\n- D_marginatus : number of *D_marginatus* collected\n- Tiques ? : sum of ticks collected
#> 4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         This document presents :\n\n- Description of tick data collected from slaughterhouse cow skins in Corsica, between 2017 and 2019,\n- A Data dictionnary,\n- Tick data collected.
#> 5                                                                                                                                                                                                                                                                                                                                                              This document presents :\n\n- Description of tick data collected in slaughterhouse in Corsica between 2020 and 2021\n- "Dictionnaire" : A data dictionnary\n- " Bovin base de données" : Tick data collected \n- "Bovin pool" : Information about tick species collected\n- "Plaque 1 à 5" : *to complete*\n- "SA 2021" : *to complete*\n- "feuille travail" : *to complete*\n- "Sanglier" :*to complete*\n
#> 6                                                                                                                    On this document, there is only the tick data collection.\n\nSome information about variable :\n\n- SIT_EXP_CODE : Sampling site code\n- Commune : Site municipality\n- Long : GPS point longitude\n- Lat : GPS point latitude\n- Date : Sampling date \n- H_marg_nb : Number of *Hyalomma Marginatum* collected\n_ H_marg_012 : Presence (1), absence (0) or unique presence (2) of *Hyalomma Marginatum*\n_ nb_cv_infest : Number of horses infested with ticks\n_ nb_cv_hyalomma : Number of horses infested with *Hyalomma Marginatum*\n_ Nb_CV_exam : Number of horses examined\n_ Traitement : treatment by the breeder less than one month ago
#> 7                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        There are on this document :\n\n- Description of tick data collected from equestrian structures in Corsica, between 2018 and 2020.\n- A Data dictionnary,\n- Tick data collected.
#> 8                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    NIRS spectrum of herbarium collection
#> 9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Résultats des collectes de moustiques, par site, localité, date et espèce.\r\nMosquito collections results per site, locality, date and species.
#>                              unf
#> 1                           <NA>
#> 2 UNF:6:kt94ifK0zK3B4KPE6x+j5g==
#> 3 UNF:6:lt7ZJ1diuShhMCd8UWq5zQ==
#> 4                           <NA>
#> 5 UNF:6:8jLV6xbX6g9YzZxfFTynzA==
#> 6                           <NA>
#> 7                           <NA>
#> 8                           <NA>
#> 9 UNF:6:E4qG20NvrCTE/jwa/yQhTg==
```

<sup>Created on 2022-02-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by pdurbin** [comment]
> I'm not sure how to pass the API token with curl. I'll check.

You can pass it as a header or a query parameter. Please see https://guides.dataverse.org/en/5.9/api/auth.html

**Comment by famuvie** [comment]
Sorry, I made a mistake in the previous example and have just corrected it. It actually works!

**Comment by famuvie** [comment]
Still, I can't find a hacky way for adding "_draft" to the file id. I guess that needs to be fixed in the package :)

**Comment by kuriwaki** [comment]
Thanks @famuvie for creating an issue. A partial fix is now on dev. 
@pdurbin, thanks for pointing out entityId. I implemented it on dev as there seems to be no downside.  

I created a test dataset on demo dataverse that is intentionally unpublished. The get commands seem to go ok except for my unpublished test file does not have a `UNF` under the SEARCH API even though it does with the File API. Have you seen this before? 

Proper UNF detection becomes necessary since that's how it currently determines if a file is ingested or not. 

```
> str(dataset_files(dataset = "10.70122/FK2/4XHVAP", server = "demo.dataverse.org")[[1]]$dataFile)
List of 16
 $ id                 : int 1951382
 $ persistentId       : chr ""
 $ pidURL             : chr ""
 $ filename           : chr "mtcars.tab"
 $ contentType        : chr "text/tab-separated-values"
 $ filesize           : int 1713
 $ storageIdentifier  : chr "s3://demo-dataverse-org:17f75571af3-60325bcbb1f1"
 $ originalFileFormat : chr "text/csv"
 $ originalFormatLabel: chr "Comma Separated Values"
 $ originalFileSize   : int 1700
 $ originalFileName   : chr "mtcars.csv"
 $ UNF                : chr "UNF:6:KRE/AItWGJWd5tJ+bboN7A=="
 $ rootDataFileId     : int -1
 $ md5                : chr "c502359c26a0931eef53b2207b2344f9"
 $ checksum           :List of 2
  ..$ type : chr "MD5"
  ..$ value: chr "c502359c26a0931eef53b2207b2344f9"
 $ creationDate       : chr "2022-03-10"
```
```
> str(dataverse_search(entityId = 1951382, server = "demo.dataverse.org", key = Sys.getenv("DATAVERSE_KEY")))
1 of 1 result retrieved
'data.frame':	1 obs. of  13 variables:
 $ name                 : chr "mtcars.csv"
 $ type                 : chr "file"
 $ url                  : chr "https://demo.dataverse.org/api/access/datafile/1951382"
 $ file_id              : chr "1951382"
 $ file_type            : chr "Comma Separated Values"
 $ file_content_type    : chr "text/csv"
 $ size_in_bytes        : int 1700
 $ md5                  : chr "c502359c26a0931eef53b2207b2344f9"
 $ checksum             :'data.frame':	1 obs. of  2 variables:
  ..$ type : chr "MD5"
  ..$ value: chr "c502359c26a0931eef53b2207b2344f9"
 $ dataset_name         : chr "Permanent draft dataset for testing"
 $ dataset_id           : chr "1951381"
 $ dataset_persistent_id: chr "doi:10.70122/FK2/4XHVAP"
 $ dataset_citation     : chr "Kuriwaki, Shiro, 2022, \"Permanent draft dataset for testing\", https://doi.org/10.70122/FK2/4XHVAP, Demo Datav"| __truncated__
```

**Comment by kuriwaki** [comment]
Yes! It was sufficient to add a data description to the draft dataset, and it somehow updated. Thank you.

**Comment by pdurbin** [comment]
@kuriwaki hmm, I can replicate this on "develop" on my laptop (around 0d853b74e9). When I first upload a file to a draft, the UNF does not appear in search results...

```
$ curl -s -H X-Dataverse-key:$API_TOKEN http://localhost:8080/api/search?q=id:datafile_5_draft | jq .
{
  "status": "OK",
  "data": {
    "q": "id:datafile_5_draft",
    "total_count": 1,
    "start": 0,
    "spelling_alternatives": {},
    "items": [
      {
        "name": "2016-06-29.csv",
        "type": "file",
        "url": "http://localhost:8080/api/access/datafile/5",
        "file_id": "5",
        "file_type": "Comma Separated Values",
        "file_content_type": "text/csv",
        "size_in_bytes": 58690,
        "md5": "d5de092a84304a9965c787b8dcd27c99",
        "checksum": {
          "type": "MD5",
          "value": "d5de092a84304a9965c787b8dcd27c99"
        },
        "dataset_name": "zzz",
        "dataset_id": "4",
        "dataset_persistent_id": "doi:10.5072/FK2/JJK8WY",
        "dataset_citation": "Admin, Dataverse, 2022, \"zzz\", https://doi.org/10.5072/FK2/JJK8WY, Root, DRAFT VERSION"
      }
    ],
    "count_in_response": 1
  }
}
```

... but if I edit the metadata of the draft dataset (forcing the file to be reindexed, the UNF appears):

```
$ curl -s -H X-Dataverse-key:$API_TOKEN http://localhost:8080/api/search?q=id:datafile_5_draft | jq .
{
  "status": "OK",
  "data": {
    "q": "id:datafile_5_draft",
    "total_count": 1,
    "start": 0,
    "spelling_alternatives": {},
    "items": [
      {
        "name": "2016-06-29.tab",
        "type": "file",
        "url": "http://localhost:8080/api/access/datafile/5",
        "file_id": "5",
        "file_type": "Tab-Delimited",
        "file_content_type": "text/tab-separated-values",
        "size_in_bytes": 59208,
        "md5": "d5de092a84304a9965c787b8dcd27c99",
        "checksum": {
          "type": "MD5",
          "value": "d5de092a84304a9965c787b8dcd27c99"
        },
        "unf": "UNF:6:6YVg+pUWsYD52stDkZuzUA==",
        "dataset_name": "zzzyyy",
        "dataset_id": "4",
        "dataset_persistent_id": "doi:10.5072/FK2/JJK8WY",
        "dataset_citation": "Admin, Dataverse, 2022, \"zzzyyy\", https://doi.org/10.5072/FK2/JJK8WY, Root, DRAFT VERSION, UNF:6:6YVg+pUWsYD52stDkZuzUA== [fileUNF]"
      }
    ],
    "count_in_response": 1
  }
}
```

Please feel free to open an issue about this at https://github.com/IQSS/dataverse/issues if you'd like.

**Comment by kuriwaki** [comment]
I will put a tip about this in the dataverse download vignette. I think it is a limitation that might be common to people who try to download draft datasets, but the current method to edit something seems not too onerous.

**Comment by kuriwaki** [comment]
Addressed by 0.3.11.
